Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -173,13 +182,16 @@ class TIFFLoadingContext {
             // From description of PhotometricInterpretation in Section 8: Baseline Field Reference Guide
             // "In a TIFF ColorMap, all the Red values come first, followed by the Green values,
             //  then the Blue values."
-            auto const size = 1 << (*m_metadata.bits_per_sample())[0];
-            auto const red_offset = 0 * size;
-            auto const green_offset = 1 * size;
-            auto const blue_offset = 2 * size;
+            u32 const size = 1 << (*m_metadata.bits_per_sample())[0];
+            u32 const red_offset = 0 * size;
+            u32 const green_offset = 1 * size;
+            u32 const blue_offset = 2 * size;
 
             auto const color_map = *m_metadata.color_map();
 
+            if (blue_offset + index >= color_map.size())
```

Comment:
Am I right in the assumption that you've widened the types above and that this cannot overflow?

Reply:
I did it to change the sign `u32` instead of `i32` allows us to compare it to `color_map.size()`
Bad news, it can already overflow here `u32 const size = 1 << (*m_metadata.bits_per_sample())[0];` :upside_down_face:

Reply:
Nothing what a quick cast to u64 at the right place can't fix?

Reply:
Oh wait even better, range checking the shift

Reply:
I'm actually more worried of `TRY(stream.read_bits<u16>(32));` being allowed

Reply:
Ok let's use `u64` for now